src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.Cache.cs

```diff
@@ -51,27 +52,29 @@ public abstract partial class JsonTypeInfo
 
         internal Func<JsonParameterInfoValues[]>? CtorParamInitFunc;
 
-        internal JsonPropertyInfo CreateProperty(Type propertyType, JsonConverter converter)
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        internal JsonPropertyInfo CreatePropertyUsingReflection(Type propertyType, JsonConverter converter)
         {
-            // Create the JsonPropertyInfo instance.
-            JsonPropertyInfo jsonPropertyInfo = converter.CreateJsonPropertyInfo(declaringTypeInfo: this, propertyType, Options);
-            jsonPropertyInfo.DefaultConverterForType = converter;
-            jsonPropertyInfo.ConverterStrategy = converter.ConverterStrategy;
-            return jsonPropertyInfo;
-        }
+            Debug.Assert(propertyType.IsInSubtypeRelationshipWith(converter.TypeToConvert));
```

Comment:
so in order to remove that bool variable we added RequiresUnreferencedCode? Is that actually worth it?

Reply:
Not exactly, it's being added 1) to ensure the `T` type of the generic `JsonPropertyInfo<T>` matches the property type and 2) because we need to root the default converters when creating a new property -- this can be necessary in cases where we're modifying sourcegen metadata where no other rooting opportunity has presented itself.

Reply:
Not sure I follow? We already have a bunch of those.

Reply:
I'm still not sure I understand this - why did we not need RUC before and we do now? If there is scenario this is fixing can you create a test for that?

Reply:
Take a look at lines 68-70 just below this comment. It needs reflection to create `JsonPropertyInfo<T>` when we're being fed a converter that is not `JsonConverter<T>`.

Reply:
I understand that's the case now but I do not understand how did we somehow not need RUC before

Reply:
I mean, did we suppress it or what?